### ISSUE TITLE
Is it possible to refute absolute truth? If so, how can this be done without proving at least one absolute truth?

### ISSUE THREAD

**Issue by codylindley** [raganwald/AMA#3]

**Comment by shawndumas** [comment]
@codylindley what is often meant by the phrase "absolute truth" is any given person knowing any given thing absolutely.

What is being denied is the Dunning-Krueger'sque insistence from the person demanding that we do something other that what we want, in that moment, to do.

In a moralistic and philosophic vacuum most are willing to admit to the notion of truth. In fact, they rely on it by pointing out the truth of the uncertainty of truth.

**Comment by codylindley** [comment]
@shawndumas - Would absolute truth cease to exist if all human brains we're gone?

**Comment by shawndumas** [comment]
@codylindley good question. 

I hold that truth is:
1. extent
2. absolute
3. eternal
4. immutable
5. propositional

So, without question, I believe truth to be free from human dependence.

**Comment by raganwald** [comment]
I use words like “truth,” but what I personally believe is that the universe as I perceive it is _consistent_.

**Comment by shawndumas** [comment]
@raganwald as in non-contradictory?

**Comment by mxriverlynn** [comment]
"which truth? whose truth?" - (some spy novel i read years ago)

**Comment by raganwald** [comment]
Exactly. And when we find a contradiction, we peel back a layer of reality and discover a deeper reality that is consistent. Until we discover another inconsistency.

**Comment by raganwald** [comment]
> "which truth? whose truth?”

Also could have been said by [Sir Humphrey Appleby](https://en.wikipedia.org/wiki/Humphrey_Appleby)

**Comment by shawndumas** [comment]
@raganwald sounds like you might be a classic Aristotelian logician.
1. `A===A`: the law of identity
2. `!A || A`: the law of the excluded middle
3. `!A !== A`: the law of non-contradiction 

I believe all points to be `true`.

**Comment by mxriverlynn** [comment]
"I believe there are no absolute truths"

"Are you sure of that?"

"Absolutely"

/me ducks and runs away :)

**Comment by davidtheclark** [comment]
It is ridiculously uncanny that I came across this less than three minutes after reading [_What's the Use of Truth?_](http://www.amazon.com/Whats-Use-Truth-Pascal-Engel/dp/0231140142/).

**Comment by codylindley** [comment]
@shawndumas Would it be certain than that truth is not subjective (relative to a human mind) and irrefutable is the nature of absolute truth? To refute an absolute truth (X is true for all people at all times) would require a denial or refuting of the law of non-contradiction, right? Which, again, proves the law of non-contradiction because one can't deny anything without it.

**Comment by codylindley** [comment]
@raganwald - What exactly is constant? I thought the steady state theory as of today was dead, right? But, you might not be using these words like I think. If anything, we factual know the universe is not constant (an absolute truth that has been true for all people at all times, regardless how they think about it). Can you elaborate?

**Comment by codylindley** [comment]
@davidtheclark I raise you one pascal. 

“Truth is so obscure in these times, and falsehood so established, that, unless we love the truth, we cannot know it.”

― Blaise Pascal

**Comment by codylindley** [comment]
@derickbailey - This is sort of the point of my question. "which truth? whose truth?" pre-suppose a truth. One can't ask such questions without first, acknowledging an absolute truth is possible. The alternative, is skepticism. And even that, is illogical, because one would have to be skeptical about the fact that there is only skepticism.

**Comment by shawndumas** [comment]
@codylindley I'd take it further. 

Language itself is impossible apart from it. [Here is a comment I wrote on HN](https://news.ycombinator.com/item?id=7715516) that bares that out.

**Comment by codylindley** [comment]
@shawndumas - It would seem that the basic laws of logic, as you stated them, are primordial. These laws exist so that anything at all can (including linguistics) . They are, absolutely true for all people at all times. They are the constant. I'm wondering if this is the constant that @raganwald was referring too. Anytime we tear anything down, there we find the laws of logic. Which result in an absolute truth. I'm trying to find someone who can refute this. BTW Aristotle logic, did Aristotle define that or discover it?

**Comment by shawndumas** [comment]
@codylindley by his own account it was a discovery; akin to how an astronomer discovers a comet.

**Comment by codylindley** [comment]
@shawndumas - I believe we have an accord. My question for @raganwald genuinely seeks to understand the cool-aid the majority of developers are drinking which results in a relativistic post modern pluralistic thinking process. We, as developers operate in a world of absolutes. Yet most developers stop thinking in this manner when it comes to reality.

**Comment by shawndumas** [comment]
@codylindley next time you have that conversation with a developer ask them to create a toy simulation -- in code -- of what they are espousing.

**Comment by raganwald** [comment]
FYI, the literal meaning of the phrase “drinking the kool-aid:”

https://en.wikipedia.org/wiki/Drinking_the_Kool-Aid

**Comment by codylindley** [comment]
Yes. I literally meant, "unquestioned belief, argument, or philosophy without critical examination".

**Comment by thinkhuman** [comment]
There seems to be a lot of confusion here between "truth" and "belief". See: Plato. And, oh, every philosopher in history. http://plato.stanford.edu/entries/truth/

Short answer? Have a beer and forget it. Claiming to know truth is like everybody hunting in a dark room for a black hat that isn't there and everyone at once yells "I found it!"

**Comment by codylindley** [comment]
@thinkhuman - Do you believe that assertion to be true? Odd, how you illustrate that point from a perspective of having the truth about the dark room.

**Comment by thinkhuman** [comment]
@codylindley I believe my assertion that you should have a beer is true. But, as Socrates said "I am often mistaken". Which is second to his most famous statement: "I drank *what*?"

**Comment by codylindley** [comment]
I'll be drinking Truth IPA http://www.rhinegeist.com/beers.
